lib/web_ui/lib/src/engine/platform_dispatcher.dart

```diff
@@ -1245,68 +1207,58 @@ bool _handleWebTestEnd2EndMessage(MethodCodec codec, ByteData? data) {
   return false;
 }
 
-/// Invokes [callback] inside the given [zone].
-void invoke(void Function()? callback, Zone? zone) {
+/// Invokes the zoned [callback].
+void invoke(ui.ZonedCallback<ui.VoidCallback>? callback) {
```

Comment:
We could also make `invoke`, `invoke1` and `invoke2` extension methods on `ZonedCallback`. E.g:
```dart
typedef ZonedCallback<T extends Function> = ({Zone zone, T fn});

extension ZonedCallbackExtension on ZonedCallback {
  void invoke() {
    if (identical(zone, Zone.current)) {
      callback();
    } else {
      zone.runGuarded(fn);
    }
  }

  // ...
}
```
then callsites can simply do:
```dart
_onDrawFrame?.invoke();
_onPlatformMessage?.invoke3(...);
```